adala/environments/kafka.py

```diff
@@ -95,12 +95,9 @@ async def message_sender(
     ):
         record_no = 0
         try:
-            for record in data:
-                await producer.send(topic, value=record)
-                record_no += 1
-                # print_text(f"Sent message: {record} to {topic=}")
+            await producer.send_and_wait(topic, value=data)
             logger.info(
-                f"The number of records sent to topic:{topic}, record_no:{record_no}"
+                f"The number of records sent to topic:{topic}, record_no:{len(data)}"
```

Comment:
should the input producer in the fastapi endpoint also have this logic to send_and_wait a whole batch at once?

Reply:
for some reason there has never been a single issue observed from the submit_batch endpoint producer -> agent , so I would say no for now